api/src/main/java/ca/bc/gov/educ/api/gradbusiness/service/v2/DistrictService.java

```diff
@@ -0,0 +1,119 @@
+    public ResponseEntity<byte[]> getSchoolReportPDFByDistcode(String distCode, String type) {
+        Calendar cal = Calendar.getInstance(TimeZone.getTimeZone("PST"), Locale.CANADA);
+        int year = cal.get(Calendar.YEAR);
+        String month = String.format("%02d", cal.get(Calendar.MONTH) + 1);
+        try {
+            Optional<District> districtDetails = Optional.ofNullable(getDistrictDetails(distCode));
+            if (districtDetails.isEmpty()) {
+                return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+            }
+            String districtId = districtDetails.get().getDistrictId();
+            var result = restService.get(String.format(educGraduationApiConstants.getSchoolReportByDistrictIdAndReportType(), districtId, type), InputStreamResource.class);
+            byte[] response = new byte[0];
+            if (result != null) {
+                response = result.getInputStream().readAllBytes();
+            }
+
+            return handleBinaryResponse(response, EducGradBusinessUtil.getFileNameSchoolReportsForDistrict(distCode,year,month,type, MediaType.APPLICATION_PDF), MediaType.APPLICATION_PDF);
+        } catch (Exception e) {
+            logger.error("Error getting school report PDF by distCode: {}", e.getMessage());
+            return getInternalServerErrorResponse(e);
+        }
+    }
+
+    /**
+     * Gets internal server error response.
+     *
+     * @param t the t
+     * @return the internal server error response
+     */
+    protected ResponseEntity<byte[]> getInternalServerErrorResponse(Throwable t) {
+        ResponseEntity<byte[]> result;
+
+        Throwable tmp = t;
+        String message;
+        if (tmp.getCause() != null) {
+            tmp = tmp.getCause();
+            message = tmp.getMessage();
+        } else {
+            message = tmp.getMessage();
+        }
+        if (message == null) {
+            message = tmp.getClass().getName();
+        }
+
+        result = ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(message.getBytes());
+        return result;
+    }
+
+
+    private ResponseEntity<byte[]> handleBinaryResponse(byte[] resultBinary, String reportFile, MediaType contentType) {
+        ResponseEntity<byte[]> response;
+        if (resultBinary.length > 0) {
+            String fileType = contentType.getSubtype().toUpperCase();
+            logger.debug("Sending {} response {} KB", fileType, resultBinary.length / (1024));
+            HttpHeaders headers = new HttpHeaders();
+            headers.add("Content-Disposition", "inline; filename=" + reportFile);
+            response = ResponseEntity
+                    .ok()
+                    .headers(headers)
+                    .contentType(contentType)
+                    .body(resultBinary);
+        } else {
+            response = ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+        }
+        return response;
+    }
+
+    public District getDistrictDetails(String distNo) {
```

Comment:
This can stay in the DistrictService class. It appropriate here, but should not be in the service with the other report code